types/vote.go

```diff
@@ -14,6 +14,9 @@ import (
 
 const (
 	nilVoteStr string = "nil-Vote"
+
+	// The maximum supported number of bytes in a vote extension.
+	MaxVoteExtensionSize int = 1024
```

Comment:
This seems likely to be too small. I thought some of the motivating cases for these wanted fairly substantial blobs (e.g., pricing oracle data).

Keeping a small default makes sense, but I wonder if we shouldn't allow it to vary.

Reply:
Let's make it this bigger for the moment (e.g. 1MiB in this PR), then we could add a `toml` param... or even better a `ConsensusParam` that allows it to vary consistently across all validators.
This param could be introduced after cutting an RC... or even better at the request of a user (I'd leave an issue open though, to show we've thought about it in advance).
The reason why I set this so small in this code, is that I was playing around doing some manual tests with this limit. Then, I forgot to set a "realistic" value.

Reply:
Yes, I think if it's going to be a parameter it would have to be a consensus param. If different nodes used different limits it seems like some nodes might accept extensions others would reject even holding the data constant.

Reply:
Agreed, having this as a consensus param makes the most sense. Perhaps in @williambanfield's follow-up to this PR where we turn vote extensions on through consensus params?

Reply:
I don't think we need to gate a release candidate on it. I do expect folks will bring it up though. 😅

Reply:
@thanethomson A consensus parameter seems like the right approach for this value, although, before we add the knob, I think we should briefly consider if it's necessary. The knob would cause Tendermint to validate the ingested data to ensure it's not larger than the value. Since we're handing the value to the application, is there an advantage to having the Tendermint process control this validation since the application can enforce any validation it likes?

Reply:
@williambanfield's point is an interesting one. We might shift responsibility to the App (which entails adding a property to the spec, not a big deal).
However, can we detach this discussion from the PR? I'd just increase the max size for the time being so that it's not a problem for integration, then we can deal with the max size in parallel.

Reply:
I'm sympathetic to the desire to avoid a knob. But if we want to do that, I think we have to be pretty generous with the default size—e.g., probably an extension should be permitted to be at least as big as a transaction.

Having _some_ default as a safety measure makes a lot of sense, but I think it needs to be a true safety value and not a tight bound. (The goal in that case is to keep the node from blowing up, not to save space)

Reply:
Thanks @cmwaters for chiming in. I wasn't aware of that. Maybe that's the simplest answer: let the envelope size limit tare care (indirectly) of the maximum extension size.